// @colyseus/testing/build/TestServer.mjs
// packages/testing/src/TestServer.ts
import { matchMaker } from "@colyseus/core";
import { ColyseusSDK } from "@colyseus/sdk";
import * as httpie from "httpie";
var ColyseusTestServer = class {
  constructor(server) {
    this.server = server;
    const hostname = "127.0.0.1";
    const port = server["port"];
    this.sdk = new ColyseusSDK(`ws://${hostname}:${port}`);
    const httpEndpoint = `http://${hostname}:${port}`;
    this.http = {
      ["get"]: (segments, opts) => httpie.get(`${httpEndpoint}${segments}`, opts),
      ["post"]: (segments, opts) => httpie.post(`${httpEndpoint}${segments}`, opts),
      ["patch"]: (segments, opts) => httpie.patch(`${httpEndpoint}${segments}`, opts),
      ["delete"]: (segments, opts) => httpie.del(`${httpEndpoint}${segments}`, opts),
      ["put"]: (segments, opts) => httpie.put(`${httpEndpoint}${segments}`, opts)
    };
  }
  // Implementation
  async createRoom(roomName, clientOptions = {}) {
    const room = await matchMaker.createRoom(roomName, clientOptions);
    return this.getRoomById(room.roomId);
  }
  connectTo(room, clientOptions = {}) {
    return this.sdk.joinById(room.roomId, clientOptions);
  }
  // Implementation
  getRoomById(roomId) {
    return matchMaker.getLocalRoomById(roomId);
  }
  async cleanup() {
    await Promise.all(matchMaker.disconnectAll());
    await this.sdk.auth.signOut();
    const driver = this.server["driver"];
    if (driver) {
      driver.clear();
    }
  }
  async shutdown() {
    await this.server.gracefullyShutdown(false);
  }
};
export {
  ColyseusTestServer
};
